packages/financial-templates-lib/test/price-feed/BasketSpreadPriceFeed.js

```diff
@@ -0,0 +1,466 @@
+const winston = require("winston");
+
+const { toWei, toBN } = web3.utils;
+
+const { MedianizerPriceFeed } = require("../../src/price-feed/MedianizerPriceFeed");
+const { BasketSpreadPriceFeed } = require("../../src/price-feed/BasketSpreadPriceFeed");
+const { PriceFeedMock } = require("../../src/price-feed/PriceFeedMock");
+
+contract("BasketSpreadPriceFeed.js", function() {
+  let baselinePriceFeeds;
+  let experimentalPriceFeeds;
+  let denominatorPriceFeed;
+  let dummyLogger;
+  let basketSpreadPriceFeed;
+
+  beforeEach(async function() {
+    dummyLogger = winston.createLogger({
+      level: "info",
+      transports: [new winston.transports.Console()]
+    });
+  });
+  it("Update", async function() {
+    const priceFeeds = [new PriceFeedMock()];
+    baselinePriceFeeds = [new MedianizerPriceFeed(priceFeeds), new MedianizerPriceFeed(priceFeeds)];
+    experimentalPriceFeeds = [new MedianizerPriceFeed(priceFeeds), new MedianizerPriceFeed(priceFeeds)];
+    denominatorPriceFeed = new MedianizerPriceFeed(priceFeeds);
+
+    basketSpreadPriceFeed = new BasketSpreadPriceFeed(
+      web3,
+      dummyLogger,
+      baselinePriceFeeds,
+      experimentalPriceFeeds,
+      denominatorPriceFeed
+    );
+
+    await basketSpreadPriceFeed.update();
+
+    // On the basket spread's update call, `priceFeeds[0]` should have been updated once for each
+    // Medianizer price feed that it is incorporated in. This is because the basket spread price feed
+    // updates its imported medianizer price feeds.
+    // Check work: 2x for baseline update, 2x experimental udpate, 1x for denominator = 5 total.
+    assert.equal(priceFeeds[0].updateCalled, 5);
+  });
+  describe("Computing basket spreads when the spread is within the range [0,2]", function() {
```

Comment:
nit: do you want to add tests for trying out different scaled price feeds? This is important seeing we've got some non-18 decimal feeds in some of the baskets (BTC denomination, for example).

I'd say you can follow the pattern done in the liquidator for looping the tests with multiple decimals. Can use a simple config like this: `const configs = [{ priceFeedDecimals: 18 }, { priceFeedDecimals: 8 }];` as you only care about the decimals in the feed.

Reply:
The only thing that really will change however is the decimals of your denomination currency I guess. In either case I think it should still be included for different base currencies (BTC & ETH)

Reply:
I actually do a test for custom price feeds for every single section of the test [here](https://github.com/UMAprotocol/protocol/pull/2354/files#diff-bfff0b1bdb2e054a81074606bef333f1339dc44757d032b3c88370e40fa2f712R219)